src/modules/utils/directory_nix.rs

```diff
@@ -0,0 +1,52 @@
+use nix::sys::stat::Mode;
+use nix::unistd::{Gid, Uid};
+use std::fs;
+use std::os::unix::fs::MetadataExt;
+use std::os::unix::fs::PermissionsExt;
+
+pub fn is_write_allowed(folder_path: &str) -> Result<bool, &'static str> {
```

Comment:
This looks great! I wouldn't mind a doc comment just generally explaining what this function is looking for (e.g. checks permission bits against current user's effective uid + list of groups) but otherwise I think it's ready to merge.

Reply:
Done